src/librustc_resolve/diagnostics.rs

```diff
@@ -673,42 +679,31 @@ impl<'a> Resolver<'a> {
 
                         segms.push(ast::PathSegment::from_ident(ident));
                         let path = Path { span: name_binding.span, segments: segms };
-                        // the entity is accessible in the following cases:
-                        // 1. if it's defined in the same crate, it's always
-                        // accessible (since private entities can be made public)
```

Comment:
So, you can see that private candidates from the current crate were suggested by design.
If the item is private, but it's in the current crate, then it's quite possible that the user just hasn't added `pub` yet.

So I'm not sure the PR as it's written now is an improvement.

What would be an improvement is recording whether the suggestion has any inaccessible segments.
Then we can filter away suggestions with inaccessible segments if at least one suggestion with all accessible segments exists.

Also, I don't think the `under_enum` flags are necessary, `is_accessible_from` should work correctly without them.

Reply:
> If the item is private, but it's in the current crate, then it's quite possible that the user just hasn't added pub yet.

Ok, but shall we keep the PR's behavior when suggesting paths of external crates?

Reply:
This PR doesn't change any observable behavior for external paths?

Reply:
Checked again, and external paths suggestions seem correct with regard to visibility in nightly. I wasn't able to reproduce the issue [in this comment](https://github.com/rust-lang/rust/issues/26454#issuecomment-621500464) from April 30th.

Also, `under_enum` is indeed redundant.

> Then we can filter away suggestions with inaccessible segments if at least one suggestion with all accessible segments exists.

Ok, will attempt this.

Reply:
Done.

If all suggestions are inaccessible, should the message indicate it?

`consider importing one of these private items`
Or: 
`consider importing one of these items (which need exporting)`

?

Reply:
Nah, seems ok as is.